src/test/ui/impl-trait/closure-in-impl-trait.rs

```diff
@@ -0,0 +1,14 @@
+// run-pass
+#![allow(unused_must_use)]
+fn bug<T>() -> impl Iterator<Item = [(); { |x: u32| { x }; 4 }]> {
```

Comment:
We're running into issues which suggest we also want examples where the `impl Trait` is in argument position.
(As that lowers to a generic parameter, which currently *isn't* a HIR owner)